### PR TITLE
Sc 3786/databricks move file on dbfs

### PR DESCRIPTION
**Pull request by Shamwayne**

### CODE REVIEW

File: databricks_blueprints/move_file_on_dbfs.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    from . import helpers
+    from . import errors
+    
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--access-token', dest='access_token', required=True)
+    parser.add_argument('--instance-id', dest='instance_id', required=True)
+    parser.add_argument('--source-file-name', dest='source_file_name', required=True)
+    parser.add_argument('--source-folder-name', dest='source_folder_name', required=True)
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py]: Not required. Should use `/FileStore/` if blank

File: databricks_blueprints/move_file_on_dbfs.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    parser.add_argument('--dest-file-name', dest='dest_file_name', required=True)
+    parser.add_argument('--dest-folder-name', dest='dest_folder_name', required=True)
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py]: Not required. Should use `/FileStore/` if blank

File: databricks_blueprints/move_file_on_dbfs.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    dest_file_name = args.dest_file_name
+    dest_folder_name = args.dest_folder_name
+    # create client
+    client = helpers.DatabricksClient(access_token, instance_id)
+    # create file paths
+    source_file_path = shipyard.combine_folder_and_file_name(
+        source_folder_name,
+        source_file_name
+    )
+    destination_file_path = shipyard.combine_folder_and_file_name(
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py]: `shipyard.files.`

File: databricks_blueprints/move_file_on_dbfs.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    access_token = args.access_token
+    instance_id = args.instance_id
+    source_file_name = args.source_file_name
+    source_folder_name = args.source_folder_name
+    dest_file_name = args.dest_file_name
+    dest_folder_name = args.dest_folder_name
+    # create client
+    client = helpers.DatabricksClient(access_token, instance_id)
+    # create file paths
+    source_file_path = shipyard.combine_folder_and_file_name(
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py]: `shipyard.files.`

File: databricks_blueprints/move_file_on_dbfs.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    move_response = client.post(move_endpoint, data=payload)
+    if move_response.status_code == requests.codes.ok:
+        print(f"File: {source_file_path} moved successfully to {destination_file_path}")
+    elif move_response.status_code == 401:
+        print("File: {source_file_path} does not exist")
+        sys.exit(errors.EXIT_CODE_DBFS_INVALID_SOURCE)
+    else:
+        print(f"DBFS {source_file_path} to {destination_file_path} failed.",
+              "response: {move_response.text}")
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py]: Should be an f-string. Just prints out the text as is.

**Review by blakeburch — COMMENTED** [review]
1) Can't move a file into a directory that doesn't exist. Needs to handle that on Databricks side.
2) Needs to handle all different error scenarios.
3) Needs to be able to support moving multiple files through regex.

File: databricks_blueprints/move_file_on_dbfs.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+
+def main():
+    args = get_args()
+    access_token = args.access_token
+    instance_id = args.instance_id
+    source_file_name = args.source_file_name
+    source_folder_name = args.source_folder_name
+    dest_file_name = args.dest_file_name
+    dest_folder_name = args.dest_folder_name
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py:100]: both folder name fields need to go through shipyard_utils cleaning

File: databricks_blueprints/helpers.py [hunk 7]
```diff
@@ -110,3 +120,15 @@ def start_cluster(token, instance_id, cluster_id):
               f"HTTP Status code: {start_response.status_code} ",
               f"and Response: {start_response.text}")
         sys.exit(errors.EXIT_CODE_CLUSTER_STATUS_ERRORED)
+
+
+def list_dbfs_files(client, folder_path):
+    """ retrieves a list of all the DBFS files with the names intact"""
+    params = {"path": folder_path}
+    response = client.get('/dbfs/list', params=params)
+    files_data = response.json()['files']
```
Comment (blakeburch) [inline databricks_blueprints/helpers.py]: It looks like I'm able to break this if I provide a folder path that doesn't exist. 


```
files_data = response.json()['files']
KeyError: 'files'

```
We should error handle for that appropriately. [](edited-with-Axolo)

File: databricks_blueprints/move_file_on_dbfs.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        sys.exit(errors.EXIT_CODE_DBFS_INVALID_SOURCE)
+    elif move_response.status_code == 500:
+        error_message = move_response.json()['message']
+        print(
+            f"Error trying to move to {destination_file_path} : {error_message}"
+        )
+        sys.exit(errors.EXIT_CODE_DBFS_MOVE_ERROR)
+    else:
+        print(f"DBFS {source_file_path} to {destination_file_path} failed.",
+              "response: {move_response.text} status: {move_response.status_code}")
```
Comment (blakeburch) [inline databricks_blueprints/move_file_on_dbfs.py]: Needs to be an f-string

**Review by blakeburch — CHANGES_REQUESTED** [review]
1) I cannot get the files to actually move when using regex. It keeps telling me `DBFS /FileStore/10mb_file_test.csv to /FileStore/10mb_file_test.csv failed.`

In the above instance, I provided a dest-file-name, so the name should have been enumerated due to regex matching logic.

Functionality works fine on exact match.
2) Needs to handle errors better (file doesn't exist, no files matched to, bad instance-id, etc.)

**Comment by blakeburch** [comment]
@Shamwayne As far as I can tell, the `list_dbfs_files` function is only listing files in the currently provided directory, not all subdirectories. Is this possible to achieve?

You can read from our Regex Documentation that the intended behavior is that searching for a file with regex means that the provided folder name is used as a "filter".

So providing "/FileStore/" should still look for regex matches against "/FileStore/test/folder/", "/FileStore/test/", "/FileStore/more/nested", "/FileStore/more" etc.

**Comment by Shamwayne** [comment]
Assumed that was specifically for delete functionality and not across the board

**Review by blakeburch — APPROVED** [review]
Minor changes made. Good to go.

**Pull request merged**